lib/phoenix/verified_routes.ex

```diff
@@ -238,6 +267,25 @@ defmodule Phoenix.VerifiedRoutes do
     })
   end
 
+  @type plug_opts :: any()
+  @type formatted_route :: %{
+          required(:verb) => String.t(),
+          required(:path) => String.t(),
+          required(:label) => String.t()
+        }
+
+  @type path :: [String.t()]
+
+  @doc """
+  Returns the necessary information about routes for display in `mix phx.routes`
+  """
+  @callback formatted_routes(plug_opts()) :: [formatted_route()]
+
+  @doc """
+  Returns `true` if the path can be verified, and false if not.
+  """
+  @callback verified_route?(plug_opts(), path) :: boolean()
```

Comment:
```suggestion
  @doc """
  Returns the necessary information about routes for display in `mix phx.routes`.
  
  The `plug_opts` is typically only passed when the router is mounted within
  a `Phoenix.Router`. Otherwise it defaults to `[]`.
  """
  @callback formatted_routes(plug_opts()) :: [formatted_route()]

  @doc """
  Returns `true` if the path is verified, and false if not.
  
  The `plug_opts` is typically only passed when the router is mounted within
  a `Phoenix.Router`. Otherwise it defaults to `[]`.
  """
  @callback verified_route?(plug_opts(), [String.t()]) :: boolean()
```